app/Models/Chat/Channel.php

```diff
@@ -454,12 +459,9 @@ public function receiveMessage(User $sender, ?string $content, bool $isAction =
                 $userChannel->markAsRead($message->message_id);
             }
 
-            if ($this->isPM()) {
-                if ($this->unhide()) {
-                    // assume a join event has to be sent if any channels need to need to be unhidden.
-                    (new ChatChannelEvent($this, $this->pmTargetFor($sender), 'join'))->broadcast();
-                }
+            $this->unhide();
```

Comment:
I just noticed the changes in `unhide` function may cause extra queries especially when doing batch sending.